docs/smtchecker.rst

```diff
@@ -404,16 +404,60 @@ that the assertion fails:
      | 		^^^^^^^^^^^^^^^^^
 
 
+******************
+SMTChecker Options
+******************
+
+Model Checking Engines
+======================
+
+The SMTChecker module implements two different reasoning engines, a Bounded
+Model Checker (BMC) and a system of Constrained Horn Clauses (CHC).  Both
+engines are currently under development, and have different characteristics.
+The engines are independent and every property warning states from which engine
+it came from. Notice that all the examples above with counterexamples were
+reported by CHC, the more powerful engine.
+
+By default both engines are used, where CHC runs first, and every property that
+was not proven is passed over to BMC. You can choose a specific engine via the CLI
+option ``--model-checker-engine {all,bmc,chc,none}`` or the JSON option
+``settings.modelChecker.engine={all,bmc,chc,none}``.
+
+Bounded Model Checker (BMC)
+---------------------------
+
+The BMC engine analyzes functions in isolation, that is, it does not take the
+overall behavior of the contract throughout many transactions into account when
+analyzing each function.  Loops are also ignored in this engine at the moment.
+Internal function calls are inlined as long as they are not recursive, direct
+or indirectly. External function calls are inlined if possible, and knowledge
```

Comment:
The knowledge erasure is independent of the inlining isn't it? I'd make it
"External function calls are inlined, if possible. Knowledge..."

Maybe I'm getting too nitpickey, though :-D.

Reply:
Correct